integrations/optimum/src/haystack_integrations/components/embedders/optimum_text_embedder.py

```diff
@@ -0,0 +1,201 @@
+            TensorRT requires to build its inference engine ahead of inference, which takes some time due to the model
+            optimization and nodes fusion. To avoid rebuilding the engine every time the model is loaded, ONNX Runtime
+            provides a pair of options to save the engine: `trt_engine_cache_enable` and `trt_engine_cache_path`. We
+            recommend setting these two provider options using the model_kwargs parameter, when using the TensorRT
+            execution provider. The usage is as follows:
+            ```python
+            embedder = OptimumTextEmbedder(
+                model="sentence-transformers/all-mpnet-base-v2",
+                onnx_execution_provider="TensorrtExecutionProvider",
+                model_kwargs={
+                    "provider_options": {
+                        "trt_engine_cache_enable": True,
+                        "trt_engine_cache_path": "tmp/trt_cache",
+                    }
+                },
+            )
+            ```
+        :param pooling_mode: The pooling mode to use. Defaults to None. When None, pooling mode will be inferred from
+            the model config. If not found, "mean" pooling will be used.
+            The supported pooling modes are:
+            - "cls": Perform CLS Pooling on the output of the embedding model. Uses the first token (CLS token) as text
+                representations.
+            - "max": Perform Max Pooling on the output of the embedding model. Uses max in each dimension over all
+                the tokens.
+            - "mean": Perform Mean Pooling on the output of the embedding model.
+            - "mean_sqrt_len": Perform mean-pooling on the output of the embedding model, but divide by sqrt
+                (input_length).
+            - "weighted_mean": Perform Weighted (position) Mean Pooling on the output of the embedding model. See
+                https://arxiv.org/abs/2202.08904.
+            - "last_token": Perform Last Token Pooling on the output of the embedding model. See
+                https://arxiv.org/abs/2202.08904 & https://arxiv.org/abs/2201.10005.
+        :param model_kwargs: Dictionary containing additional keyword arguments to pass to the model.
+            In case of duplication, these kwargs override `model`, `onnx_execution_provider`, and `token` initialization
+            parameters.
+        """
+        check_valid_model(model, HFModelType.EMBEDDING, token)
+        self.model = model
+
+        self.token = token
+        token = token.resolve_value() if token else None
+
+        if isinstance(pooling_mode, str):
+            self.pooling_mode = PoolingMode.from_str(pooling_mode)
+        # Infer pooling mode from model config if not provided,
+        if pooling_mode is None:
+            self.pooling_mode = HFPoolingMode.get_pooling_mode(model, token)
+        # Set default to "mean" if not found in model config and not specified by user
```

Comment:
I actually think it would be better to raise an error here if no pooling mode was provided or detected. This will still fail fast since it is in the init method and by providing a default we allow the user to make a silent error (e.g. the config download from HF could have failed due to a server issue or loss of wifi) which then would lead to sub-par results when using the embeddings.

Reply:
Makes sense! I will update it so that it raises an error when no pooling mode is provided or detected.